corehq/ex-submodules/pillowtop/processors/interface.py

```diff
@@ -7,3 +15,53 @@ class PillowProcessor(object):
     @abstractmethod
     def process_change(self, pillow_instance, change, is_retry_attempt=False):
         pass
+
+
+class ChunkedPillowProcessor(PillowProcessor):
+    def __init__(self, chunk_size=PYTHONPILLOW_CHUNK_SIZE):
+        self.change_queue = []
+        # explicitly check against None since we want to pass chunk_size=0 through
+        self.chunk_size = chunk_size if chunk_size is not None else PYTHONPILLOW_CHUNK_SIZE
+        self.use_chunking = chunk_size > 0
+        self.last_processed_time = None
+
+    def process_change(self, pillow_instance, change, is_retry_attempt=False):
+        if self.use_chunking and not is_retry_attempt:
+            self.change_queue.append(change)
+            if self.queue_full or self.wait_expired:
+                self.process_chunk()
+        else:
+            self.process_individual_change(change)
+
+    def process_chunk(self):
+        for change in self.change_queue:
+            self.process_individual_change(change)
```

Comment:
Yea good point. I didn't think to hard about this but I guess the chucking
only helps if we do bulk loads from couch. Getting rid of the chucking
would simplify a lot of things!